File: 1342_number_of_steps_to_reduce_to_zero/number_of_steps.py
def numberOfSteps(num: int) -> int:
    """
    Time: O(1)
    Space: O(1)
    """
    res = 0
    while num:
        if num & 1:
            num -= 1
        else:
            num //= 2
        res += 1
    return res
        

File: 1342_number_of_steps_to_reduce_to_zero/number_of_steps.java
class Solution {
    // Time: O(1)
    // Space: O(1)
    public int numberOfSteps (int num) {
        if (num == 0) {
            return 0;
        }
        int res = 0;
        while (num != 0) {
            res += ((num & 1) == 0 ? 1 : 2);
            num >>= 1;
                
        }
        return res - 1;
    }
}